### PR TITLE
Issue/6866 log version startup

### LINKED ISSUE #6866: Log the version of the inmanta-server at startup

### PR DESCRIPTION
**Pull request by FloLey**
# Description

log server version at startup

closes #6866

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: src/inmanta/app.py [hunk 1]
```diff
@@ -83,6 +83,7 @@
 
 @command("server", help_msg="Start the inmanta server")
 def start_server(options: argparse.Namespace) -> None:
+    LOGGER.info(f"Starting inmanta-server version {__version__}")
```
Comment (arnaudsjs) [inline src/inmanta/app.py:86]: This will log the version of the inmanta-core component. The idea was to log the version of the product.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: src/inmanta/app.py [hunk 2]
```diff
@@ -83,6 +83,7 @@
 
 @command("server", help_msg="Start the inmanta server")
 def start_server(options: argparse.Namespace) -> None:
+    LOGGER.info(f"Starting inmanta-server version {COMPILER_VERSION}")
```
Comment (arnaudsjs) [inline src/inmanta/app.py:86]: The compiler version and the product version are not the same thing. The log line should contains the product version. This is the version of the product python package. For ISO the product package is called `inmanta-service-orchestrator`. For OSS it's called `inmanta`.

Comment (arnaudsjs) [inline src/inmanta/app.py:86]: Make sure to use the `%s` notation.
```suggestion
    LOGGER.info(f"Starting inmanta-server version %s", COMPILER_VERSION)
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: src/inmanta/app.py (hunk 2 above)
Comment (FloLey) [reply src/inmanta/app.py]: Ow now I see, thanks

File: src/inmanta/app.py [hunk 3]
```diff
@@ -81,8 +82,30 @@
...
+    """
+    try:
+        # Try to get the version of 'inmanta-service-orchestrator' (ISO)
+        return importlib.metadata.version("inmanta-service-orchestrator")
+    except importlib.metadata.PackageNotFoundError:
+        # If not found, try 'inmanta' (OSS)
+        try:
+            return importlib.metadata.version("inmanta")
+        except importlib.metadata.PackageNotFoundError:
+            return
```
Comment (arnaudsjs) [inline src/inmanta/app.py:100]: * I notice that this code is more or less a duplication of the code in the `FeatureManager` class (`FeatureManager.get_product_metadata().version`). I think we should rely on that logic instead.
* I would also move this log line into the `InmantaBootloader.start()` method. That way the log line is also present when the server is started in a different way than using the `inmanta` command.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**